### ISSUE TITLE
PolygonSockClient - request to subscribe for many symbols

### ISSUE THREAD

**Issue by ooples** [alpacahq/alpaca-trade-api-csharp#117]
I know this is based on the polygon api but you can subscribe to multiple stock symbols at the same time. In the examples I can see here, it looks like it was only designed for one symbol at a time unless I'm missing something. Is there an example available for subscribing for something like 10 symbols from a portfolio to check for any changes?

**Comment by OlegRa** [comment]
@ooples You can test new methods for batch subscription/unsubscription in [3.3.0-alpha](https://www.nuget.org/packages/Alpaca.Markets/3.3.0-alpha) pre-release version. If all work as expected, please, close this issue.

**Comment by ooples** [comment]
I have a few questions. If you have multiple open positions ie a portfolio, will these positions be automatically subscribed to? Also will you add a feature that can allow us to use backtest strategies with paper trading using historical data. We would use the order method with a custom parameter for date info so we can make past trades. Unless this is already possible and I'm not aware?

**Comment by OlegRa** [comment]
This SDK is just a thin wrapper around HTTP/WebSocket APIs and nothing more. Each method call is independent and your code is responsible to properly subscribe/unsubscribe contracts which required in your biz logic. Same is true about 'past trades' - Alpaca API itself doesn't support such type of orders so SDK doesn't provide them too. You have to implement them yourself (or use another existing library for it) but you can use historical data provided by this SDK for calculating position values like P&L. I'm thinking now about more high-level library which will hide all API details behind more simple trading-oriented API but this idea still in development.

**Comment by ooples** [comment]
From my testing it looks all of the connection issues I had with the websockets is actually because of these new methods. I'm going to need you to verify on your end but I have run tests on 20 symbols at a time, 100 symbols, 200 symbols and the results are all over the place. When I run 20 symbols at a time it only gets results from around 7 or so at  a time but that changes sporadically. Same for 100 or 200 but when I do anything over that, it competely freezes up and that is when it closes the socket and crashes.

**Comment by OlegRa** [comment]
I'm unable to reproduce this problem on Polygon.io stage environment even if I try 50 contracts in one subscription. Please, provide your contracts list and I'll investigate problem on my side ASAP.

**Comment by ooples** [comment]
I don't believe it is the list itself that is the issue. I ran a test and commented out all of the code in the minute received event and it worked as expected. I then checked several websocket properties and saw that the websocket status goes to None and I changed the ping interval to the default 60 but I'm still seeing these issues

**Comment by OlegRa** [comment]
Could you please attach here minimal project which reproduces issue with disconnected web socket.

**Comment by ooples** [comment]
I would prefer to email it to you

**Comment by OlegRa** [comment]
You can find my email on GitHub profile page.

**Comment by ooples** [comment]
@OlegRa I emailed it you. Please confirm that you have received it

**Comment by OlegRa** [comment]
@ooples I'm unable to download this archive from Google Drive. Minimal code for reproducing problem should be simple one file - just share it as Gist.

**Comment by ooples** [comment]
@OlegRa I published it on a github private repo and the link is https://github.com/ooples/alpacaexample

I included my original code because I wanted you to be able to get the same symbols that I'm using to reproduce the issue. I'm not sure how to share my code with you though. I have never done that before

**Comment by OlegRa** [comment]
@ooples OK. I've checked you code and I have lot of questions about it. But first - I'm unable to reproduce problem with disconnected web socket. I've run your code for a while and all works fine on my side - it connected and data for all contracts are received after one minute.
So for me this issue is closed. If you want my support with your code - we can discuss it in scope of separate issue in your repository (I've already created it).
